OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,527 @@
+		public readonly int WindStrength;
+		
+		[Desc("The first color of the snow particles. The default is the 128th. Color from the temperate.pal")]
+		public readonly Color Snow1 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 236, 236, 236);
+
+		[Desc("The second color of the snow particles. The default is the 129th. Color from the temperate.pal")]
+		public readonly Color Snow2 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 228, 228, 228);
+
+		[Desc("The third color of the snow particles. The default is the 130th. Color from the temperate.pal")]
+		public readonly Color Snow3 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 208, 208, 208);
+
+		[Desc("The fourth color of the snow particles. The default is the 131st. Color from the temperate.pal")]
+		public readonly Color Snow4 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 188, 188, 188);
+
+
+		[Desc("The first color of the rain particles. The default is the 164th. Color from the temperate.pal")]
+		public readonly Color Rain1 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 160, 168, 200);
+
+		[Desc("The second color of the rain particles. The default is the 165th. Color from the temperate.pal")]
+		public readonly Color Rain2 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 140, 148, 184);
+
+		[Desc("The third color of the rain particles. The default is the 166th. Color from the temperate.pal")]
+		public readonly Color Rain3 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 124, 132, 172);
+
+		[Desc("The fourth color of the rain particles. The default is the 167th. Color from the temperate.pal")]
+		public readonly Color Rain4 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 104, 116, 160);
+
+
+		[Desc("The first color of the d2kSand particles. The default is the 78th. Color from the d2k.pal")]
+		public readonly Color d2kSand1 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 248, 196, 132);
+
+		[Desc("The second color of the d2kSand particles. The default is the 82th. Color from the d2k.pal")]
+		public readonly Color d2kSand2 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 236, 176, 112);
+
+		[Desc("The third color of the d2kSand particles. The default is the 85th. Color from the d2k.pal")]
+		public readonly Color d2kSand3 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 204, 136, 68);
+
+		[Desc("The fourth color of the d2kSand particles. The default is the 89th. Color from the d2k.pal")]
+		public readonly Color d2kSand4 = Color.FromArgb(255 - Game.CosmeticRandom.Next(0, 51), 188, 116, 52);
+		
+		public object Create(ActorInitializer init) { return new WeatherOverlay(init.World, this); }
+	}
+
+	public class WeatherOverlay : ITick, IPostRender
+	{
+		public bool Enabled = true;
+		readonly WeatherOverlayInfo info;
+		readonly World world;
+		public enum WeatherVariants { snow, rain, d2kSand };
+		public struct Particle
```

Comment:
This class has a huge amount of state. Is it possible to reduce the amount of variables you use?